diff --git a/src/applications/differential/storage/DifferentialDiff.php b/src/applications/differential/storage/DifferentialDiff.php
index 8976fdd..4667e34 100644
--- a/src/applications/differential/storage/DifferentialDiff.php
+++ b/src/applications/differential/storage/DifferentialDiff.php
@@ -142,7 +142,7 @@ final class DifferentialDiff
       DifferentialDefaultViewCapability::CAPABILITY);
 
     $diff = id(new DifferentialDiff())
-      ->setViewPolicy($view_policy);
+      ->setViewPolicy($actor->getPHID());
 
     return $diff;
   }
