diff --git a/src/applications/feed/conduit/FeedQueryConduitAPIMethod.php b/src/applications/feed/conduit/FeedQueryConduitAPIMethod.php
index bddc4f592..7ada44947 100644
--- a/src/applications/feed/conduit/FeedQueryConduitAPIMethod.php
+++ b/src/applications/feed/conduit/FeedQueryConduitAPIMethod.php
@@ -1,6 +1,6 @@
 <?php
 
-final class FeedQueryConduitAPIMethod extends FeedConduitAPIMethod {
+class FeedQueryConduitAPIMethod extends FeedConduitAPIMethod {
 
   public function getAPIMethodName() {
     return 'feed.query';
@@ -54,7 +54,7 @@ final class FeedQueryConduitAPIMethod extends FeedConduitAPIMethod {
     return 'nonempty dict';
   }
 
-  protected function execute(ConduitAPIRequest $request) {
+  public function execute(ConduitAPIRequest $request) {
     $results = array();
     $user = $request->getUser();
 
diff --git a/src/applications/feed/query/PhabricatorFeedQuery.php b/src/applications/feed/query/PhabricatorFeedQuery.php
index ae1da3aba..2e5efb93b 100644
--- a/src/applications/feed/query/PhabricatorFeedQuery.php
+++ b/src/applications/feed/query/PhabricatorFeedQuery.php
@@ -1,6 +1,6 @@
 <?php
 
-final class PhabricatorFeedQuery
+class PhabricatorFeedQuery
   extends PhabricatorCursorPagedPolicyAwareQuery {
 
   private $filterPHIDs;
