diff --git a/src/applications/differential/controller/DifferentialRevisionViewController.php b/src/applications/differential/controller/DifferentialRevisionViewController.php
old mode 100644
new mode 100755
index 6b13e89..05ebd30
--- a/src/applications/differential/controller/DifferentialRevisionViewController.php
+++ b/src/applications/differential/controller/DifferentialRevisionViewController.php
@@ -172,6 +172,9 @@ final class DifferentialRevisionViewController
 
     $request_uri = $request->getRequestURI();
 
+    $moz_warnings = new DifferentialRevisionWarning();
+    $moz_warnings = $moz_warnings->createWarnings($viewer, $revision);
+
     $large = $request->getStr('large');
 
     $large_warning =
@@ -584,6 +587,7 @@ final class DifferentialRevisionViewController
       ->setMainColumn(
         array(
           $operations_box,
+          $moz_warnings,
           $info_view,
           $details,
           $diff_detail_box,
