diff --git a/src/applications/differential/storage/DifferentialRevision.php b/src/applications/differential/storage/DifferentialRevision.php
old mode 100644
new mode 100755
index 2f7cee110..8ab3eebbd
--- a/src/applications/differential/storage/DifferentialRevision.php
+++ b/src/applications/differential/storage/DifferentialRevision.php
@@ -73,7 +73,7 @@ final class DifferentialRevision extends DifferentialDAO
     $view_policy = $app->getPolicy(
       DifferentialDefaultViewCapability::CAPABILITY);
 
-    if (PhabricatorEnv::getEnvConfig('phabricator.show-prototypes')) {
+    if (PhabricatorEnv::getEnvConfig('phabricator.show-prototypes') || true) {
       $initial_state = DifferentialRevisionStatus::DRAFT;
       $should_broadcast = false;
     } else {
