diff --git a/src/applications/differential/engine/DifferentialDiffExtractionEngine.php b/src/applications/differential/engine/DifferentialDiffExtractionEngine.php
index 6ffe33333..03c9875b2 100644
--- a/src/applications/differential/engine/DifferentialDiffExtractionEngine.php
+++ b/src/applications/differential/engine/DifferentialDiffExtractionEngine.php
@@ -113,7 +113,40 @@ final class DifferentialDiffExtractionEngine extends Phobject {
 
     // TODO: Attach binary files.
 
-    return $diff->save();
+    $diff->save();
+
+    $commit_data = $commit->getCommitData();
+    $message = $commit_data->getCommitMessage();
+
+    $properties = array(
+      'local:commits' => array(
+        $identifier => array(
+          'author' => $commit_data->getCommitDetail('authorName'),
+          'commit' => $identifier,
+          'rev' => $identifier,
+          'authorEmail' => $commit_data->getCommitDetail('authorEmail'),
+          'summary' => $commit_data->summarizeCommitMessage($message),
+          'message' => $message,
+        ),
+      ),
+    );
+
+    foreach ($properties as $name => $value) {
+      $property = id(new DifferentialDiffProperty())->loadOneWhere(
+        'diffID = %d AND name = %s',
+        $diff->getID(),
+        $name);
+      if (!$property) {
+        $property = id(new DifferentialDiffProperty())
+          ->setDiffID($diff->getID())
+          ->setName($name);
+      }
+      $property
+        ->setData($value)
+        ->save();
+    }
+
+    return $diff;
   }
 
   public function isDiffChangedBeforeCommit(
diff --git a/src/applications/differential/management/PhabricatorDifferentialExtractWorkflow.php b/src/applications/differential/management/PhabricatorDifferentialExtractWorkflow.php
index 3afe232c4..1b8ff4813 100644
--- a/src/applications/differential/management/PhabricatorDifferentialExtractWorkflow.php
+++ b/src/applications/differential/management/PhabricatorDifferentialExtractWorkflow.php
@@ -47,6 +47,7 @@ final class PhabricatorDifferentialExtractWorkflow
           $extract));
     }
 
+    $commit->attachCommitData($commit->loadCommitData());
     $diff = id(new DifferentialDiffExtractionEngine())
       ->setViewer($viewer)
       ->newDiffFromCommit($commit);
diff --git a/src/applications/diffusion/worker/DiffusionUpdateObjectAfterCommitWorker.php b/src/applications/diffusion/worker/DiffusionUpdateObjectAfterCommitWorker.php
index a0a18c066..462e6ce3c 100644
--- a/src/applications/diffusion/worker/DiffusionUpdateObjectAfterCommitWorker.php
+++ b/src/applications/diffusion/worker/DiffusionUpdateObjectAfterCommitWorker.php
@@ -184,7 +184,7 @@ final class DiffusionUpdateObjectAfterCommitWorker
       ->setAuthorPHID($acting_phid);
 
     $content_source = $this->newContentSource();
-
+    $commit->attachCommitData($data);
     $extraction_engine->updateRevisionWithCommit(
       $revision,
       $commit,
