diff --git a/src/applications/settings/setting/PhabricatorEmailNotificationsSetting.php b/src/applications/settings/setting/PhabricatorEmailNotificationsSetting.php
index dfbedb3a12..d5fad0617e 100644
--- a/src/applications/settings/setting/PhabricatorEmailNotificationsSetting.php
+++ b/src/applications/settings/setting/PhabricatorEmailNotificationsSetting.php
@@ -7,6 +7,7 @@ final class PhabricatorEmailNotificationsSetting

   const VALUE_SEND_MAIL = '0';
   const VALUE_NO_MAIL = '1';
+  const VALUE_MOZILLA_MAIL = '2';

   public function getSettingName() {
     return pht('Email Notifications');
@@ -22,12 +23,19 @@ final class PhabricatorEmailNotificationsSetting

   protected function getControlInstructions() {
     return pht(
-      'If you disable **Email Notifications**, Phabricator will never '.
+      ' - To receive the new Mozilla-specific emails, select **Mozilla '.
+      'Notifications**. Note that herald rules will no longer '.
+      'send you emails.'.
+      "\n".
+      ' - To continue to receive regular Phabricator emails, including Herald '.
+      'mail, choose **Legacy Phabricator Notifications**.'.
+      "\n".
+      ' - If you select **Disable Email Notifications**, Phabricator will never '.
       'send email to notify you about events. This preference overrides '.
       'all your other settings.'.
       "\n\n".
-      "//You will still receive some administrative email, like password ".
-      "reset email.//");
+      "//Regardless of this setting, you will still receive some ".
+      "administrative email, like password reset email.//");
   }

   public function getSettingDefaultValue() {
@@ -36,7 +44,8 @@ final class PhabricatorEmailNotificationsSetting

   protected function getSelectOptions() {
     return array(
-      self::VALUE_SEND_MAIL => pht('Enable Email Notifications'),
+      self::VALUE_MOZILLA_MAIL => pht('Mozilla Notifications'),
+      self::VALUE_SEND_MAIL => pht('Legacy Phabricator Notifications'),
       self::VALUE_NO_MAIL => pht('Disable Email Notifications'),
     );
   }
diff --git a/src/applications/settings/setting/PhabricatorEmailSelfActionsSetting.php b/src/applications/settings/setting/PhabricatorEmailSelfActionsSetting.php
index f910c2b039..92b39283d0 100644
--- a/src/applications/settings/setting/PhabricatorEmailSelfActionsSetting.php
+++ b/src/applications/settings/setting/PhabricatorEmailSelfActionsSetting.php
@@ -22,8 +22,8 @@ final class PhabricatorEmailSelfActionsSetting

   protected function getControlInstructions() {
     return pht(
-      'If you disable **Self Actions**, Phabricator will not notify '.
-      'you about actions you take.');
+      'If you disable **Self Actions**, Legacy Phabricator Notifications '.
+      'will not notify you about actions you take.');
   }

   public function getSettingDefaultValue() {
